core/network/src/sync.rs

```diff
@@ -452,6 +462,43 @@ impl<B: BlockT> ChainSync<B> {
 		}
 	}
 
+	/// Returns the new sync state of the peer and the block number that 
+	/// we need to request in case the sync state is other than Available.
+	fn handle_ancestor_search_state(
+		n: NumberFor<B>,
+		state: AncestorSearchState<B>,
+		block_hash_match: bool,
+	) -> (PeerSyncState<B>, NumberFor<B>) {
```

Comment:
Seems like the block number is unnecessary. You can just look at the returned `PeerSyncState` and if `AncestorSearch` extract the block number from there.